### ISSUE TITLE
API to wait on multiple kernel objects

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1501]
**_Reported by Johan Hedberg:_**

When needing to wait on multiple objects at the same time, in order to not have to create a dedicated thread for every object, an API that can take multiple objects as input is needed. There is a proposed k_poll API that fulfills this requirement. In the Bluetooth subsystem our specific requirement is to be able to wait on an arbitrary number of k_sem and k_fifo objects through a single k_poll call.

The reason this is so critical for Bluetooth is that we need a dedicated FIFO for every Bluetooth connection, each of which requires a dedicated thread with the currently available API. Considering that each of these threads have a stack size of 512 bytes or more, this simply doesn't scale well when we want to support many simultaneous connections (10, 20, in some cases even 30). The need to wait on semaphores in turn comes from HCI command and connection data packet flow control that's enforced by the Bluetooth controller.

There is a patch proposal for k_poll, but it hasn't been updated for quite some time:
https://gerrit.zephyrproject.org/r/#/c/4872/

(Imported from Jira ZEP-1645)

**Comment by nashif** [comment]
**_by Johan Hedberg:_**

Apologies if this is a duplicate of an existing item for k_poll (however I failed to find it). A bit before 1.6 was released Ben told me k_poll would get high priority for 1.7, so I assumed a Jira item also exists, but apparently that was a wrong assumption by me.

**Comment by nashif** [comment]
**_by Benjamin Walsh:_**

commit acc68c1

**Comment by nashif** [comment]
**_by Johan Hedberg:_**

Seems to work fine based on my testing.
